OpenRA.Game/Traits/Player/FrozenActorLayer.cs

```diff
@@ -129,7 +138,7 @@ public IEnumerable<IRenderable> Render(WorldRenderer wr)
 			return renderables;
 		}
 
-		public bool HasRenderables { get { return renderables.Any(); } }
+		public bool HasRenderables { get { return !Shrouded && renderables.Any(); } }
```

Comment:
I'm suggesting the check should be moved to the caller, not that it shouldn't happen.